boost.cmake

```diff
@@ -41,11 +41,53 @@ MACRO(SEARCH_FOR_BOOST)
     SET(BOOST_REQUIRED 1.40)
   ENDIF(NOT BOOST_REQUIRED)
 
+  # First try to find Boost to get the version
+  FIND_PACKAGE(Boost ${BOOST_REQUIRED})
+
   IF(NOT DEFINED BOOST_COMPONENTS)
     SET(BOOST_COMPONENTS
       filesystem system thread program_options unit_test_framework)
   ENDIF(NOT DEFINED BOOST_COMPONENTS)
 
+  # Check if python is in the list and adjust the version according to the current Python version.
+  # This is made mandatory if for Boost version greater than 1.67.0
+  IF(Boost_VERSION VERSION_GREATER "1.67.0")
+    LIST(FIND BOOST_COMPONENTS python PYTHON_IN_BOOST_COMPONENTS)
+    SET(BOOST_COMPONENTS_ ${BOOST_COMPONENTS})
+    IF(${PYTHON_IN_BOOST_COMPONENTS} GREATER -1)
+      # Check if Python has been found
+      IF(NOT PYTHONLIBS_FOUND)
+        MESSAGE(FATAL_ERROR "PYTHON has not been found. Please first check for Python.")   
+      ENDIF(NOT PYTHONLIBS_FOUND)
+
+      LIST(REMOVE_AT BOOST_COMPONENTS_ ${PYTHON_IN_BOOST_COMPONENTS})
+      SET(BOOST_PYTHON_COMPONENT "python${PYTHON_VERSION_MAJOR}${PYTHON_VERSION_MINOR}")
+      LIST(APPEND BOOST_COMPONENTS_ ${BOOST_PYTHON_COMPONENT})
+    ENDIF()
+  ELSE(Boost_VERSION VERSION_GREATER "1.67.0")
+    IF(${PYTHON_VERSION_MAJOR} EQUAL 3) 
+      LIST(FIND BOOST_COMPONENTS python PYTHON_IN_BOOST_COMPONENTS)
+      SET(BOOST_COMPONENTS_ ${BOOST_COMPONENTS})
+      LIST(REMOVE_AT BOOST_COMPONENTS_ ${PYTHON_IN_BOOST_COMPONENTS})
+      IF(${PYTHON_IN_BOOST_COMPONENTS} GREATER -1)
+        # Check if Python has been found
+        IF(NOT PYTHONLIBS_FOUND)
+          MESSAGE(FATAL_ERROR "PYTHON has not been found. Please first check for Python.")   
+        ENDIF(NOT PYTHONLIBS_FOUND)
+
+        IF(${UNIX})
+          SET(BOOST_PYTHON_COMPONENT "python-py${PYTHON_VERSION_MAJOR}${PYTHON_VERSION_MINOR}")
+        ELSE(${UNIX})
+          SET(BOOST_PYTHON_COMPONENT "python${PYTHON_VERSION_MAJOR}${PYTHON_VERSION_MINOR}")
+        ENDIF(${UNIX})
+        LIST(APPEND BOOST_COMPONENTS_ ${BOOST_PYTHON_COMPONENT})
+      ENDIF(${PYTHON_IN_BOOST_COMPONENTS} GREATER -1)
+    ELSE(${PYTHON_VERSION_MAJOR} EQUAL 3) 
+      SET(BOOST_COMPONENTS_ ${BOOST_COMPONENTS})
```

Comment:
This is assuming the default for `libboost-python.so` is going to remain 2.7 as default. I'm guessing this will change sooner than later with distributions shifting their default `python` to Python 3.

Unless the default is decided by Boost and not the distribution. In that case, the assumption is true until 1.67 right?

Reply:
This is what I read in the new FindBoost files concerning Boost 1.67.0
Anyway, the standard procedure should remain valid for Python 2.7 and Boost < 1.67.0.